enterprise/server/remote_execution/execution_server/execution_server.go

```diff
@@ -1335,9 +1345,13 @@ func (s *ExecutionServer) markTaskComplete(ctx context.Context, actionResourceNa
 		}
 	}
 
-	if err := s.updateUsage(ctx, executeResponse, properties); err != nil {
-		// TODO(vanja) should this be done when the executor got a cache hit?
-		log.CtxWarningf(ctx, "Failed to update usage for ExecuteResponse %+v: %s", executeResponse, err)
+	// Don't charge customers on retry attempts, as often that's due to infrastructure
+	// issues on our end.
+	if !auxMeta.GetRetry() {
```

Comment:
If `Retry == true`, could we increment some prometheus histogram metrics like CPU time and worker duration? This would let us track how much wasted compute there is due to retries, and how much latency we're adding to each action because of retries.